pkg/orderers/orderer.go

```diff
@@ -141,232 +142,251 @@ func NewOrdererModule(
 		logger:           logger,
 		view:             0,
 		inViewChange:     false,
-		viewChangeStates: make(map[types2.ViewNr]*pbftViewChangeState),
+		viewChangeStates: make(map[ot.ViewNr]*pbftViewChangeState),
 	}
-}
-
-func newOrdererConfig(issParams *issconfig.ModuleParams, membership []t.NodeID, epochNr tt.EpochNr) *PBFTConfig {
-
-	// Return a new PBFT configuration with selected values from the ISS configuration.
-	return &PBFTConfig{
-		Membership:               membership,
-		MaxProposeDelay:          issParams.MaxProposeDelay,
-		MsgBufCapacity:           issParams.MsgBufCapacity,
-		DoneResendPeriod:         issParams.PBFTDoneResendPeriod,
-		CatchUpDelay:             issParams.PBFTCatchUpDelay,
-		ViewChangeSNTimeout:      issParams.PBFTViewChangeSNTimeout,
-		ViewChangeSegmentTimeout: issParams.PBFTViewChangeSegmentTimeout,
-		ViewChangeResendPeriod:   issParams.PBFTViewChangeResendPeriod,
-		epochNr:                  epochNr,
-	}
-}
-
-// ApplyEvent receives one event and applies it to the PBFT Orderer state machine, potentially altering its state
-// and producing a (potentially empty) list of more events.
-func (orderer *Orderer) ApplyEvent(event *eventpb.Event) (*events.EventList, error) {
-	switch ev := event.Type.(type) {
-	case *eventpb.Event_Init:
-		return orderer.applyInit()
-	case *eventpb.Event_Crypto:
-		switch e := ev.Crypto.Type.(type) {
-		case *cryptopb.Event_SignResult:
-			return orderer.applySignResult(e.SignResult)
-		case *cryptopb.Event_SigsVerified:
-			return orderer.applyNodeSigsVerified(e.SigsVerified)
-		default:
-			return nil, fmt.Errorf("unknown crypto event type: %T", e)
+	m := dsl.NewModule(moduleConfig.Self)
+
+	eventpbdsl.UponInit(m, func() error {
+		return orderer.applyInit(m)
+	})
+
+	cryptopbdsl.UponSignResult(m, func(signature []byte, context *pbftpbtypes.ViewChange) error {
+		orderer.applyViewChangeSignResult(m, signature, context)
+		return nil
+	})
+
+	cryptopbdsl.UponSigVerified(m, func(nodeID t.NodeID, error error, context *pbftpbtypes.SignedViewChange) error {
+		// Ignore events with invalid signatures.
+		if error != nil {
+			orderer.logger.Log(logging.LevelWarn,
+				"Ignoring invalid signature, ignoring event (with all signatures).",
+				"from", nodeID,
+				"error", error,
+			)
+			return nil
 		}
-	case *eventpb.Event_Transport:
-		switch e := ev.Transport.Type.(type) {
-		case *transportpb.Event_MessageReceived:
-			return orderer.applyMessageReceived(e.MessageReceived)
-		default:
-			return nil, fmt.Errorf("unknown transport event type: %T", e)
+		// Verify all signers are part of the membership
+		if !sliceutil.Contains(orderer.config.Membership, nodeID) {
+			orderer.logger.Log(logging.LevelWarn,
+				"Ignoring SigVerified message as it contains signatures from non members, ignoring event (with all signatures).",
```

Comment:
`SigVerified` is not a message, say which corresponding message is being ignored. (Also, there is only one signature.)
That said, this code might change in the next pass anyway, so feel free to resolve these comments in this PR without addressing them, and fix it in the next PR.

Reply:
`SigVerified` is actually a message, and it is the message that is being ignored here (with a context).

Reply:
Technically, `SigVerified` is an internal event, `SignedViewChange` is a message (the former is the result of the verification of the signature inside the latter). But I'm being pedantic here, don't worry about it.